WooCommerce/Classes/ViewRelated/Blaze/CampaignCreation/BlazeCampaignCreationFormViewModel.swift

```diff
@@ -217,9 +220,51 @@ final class BlazeCampaignCreationFormViewModel: ObservableObject {
         initializeAdFinalDestination()
     }
 
+    func onAppear() {
+        if suggestions.isEmpty {
+            Task {
+                await loadAISuggestions()
+            }
+        }
+
+        if image == nil {
+            Task {
+                await downloadProductImage()
+            }
+        }
+    }
+
```

Comment:
I like that this is put in `.onAppear()` instead of called during init like in the previous commit. I was working on something similar about putting async work inside init and apparently it's not a best practice (although I'm not super sure about this).

I'm curious what's the difference between this, and putting it inside `.task { ... }` but with adding the conditionals? 

I can see how checking suggestions/image status is easier done inside the viewmodel. I wonder if it's possible to call `onAppear()` inside `.task`?

As far as I know, both `.onAppear` and `.task` gets called when navigating back, but `.task` is the more semantic way to run async tasks and has a built-in mechanism to handle task cancellation.

ref: https://medium.com/@ramdhas/task-vs-onappear-78cda282342d

Reply:
Hafiz has a good point. I think it'd be nice if we could replace `onAppear` with an async method `loadData`, and use `.task` to trigger it on the view. It's also more straightforward to test the async method than to test a sync method that contains `Task`. You would not have to use `waitUntil` in unit tests.

Reply:
I agree. Done in https://github.com/woocommerce/woocommerce-ios/pull/11771/commits/4aee735fad0332c80febc04acb03680605d2cf7c